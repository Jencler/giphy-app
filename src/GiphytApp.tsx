import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { Header } from "./components/Header";
import { GiphyGrid } from "./components/GiphyGrid";

export const GiphytApp = () => {
  const [categories, setCategories] = useState<string[]>([]);

  const handleAddCategory = (category: string) => {
    const categoryExists = categories.some(
      (existingCategory) =>
        existingCategory.toLowerCase() === category.toLowerCase()
    );

    if (categoryExists) return;

    setCategories([category, ...categories]);
  };

  return (
    <>
      <Header />
      <main className="main">
        <section className="main__top__section">
          <h1>Giphys Application ðŸŽ‰</h1>
        </section>
        <AddCategory onAddCategory={handleAddCategory} />
        <section className="main__giphy-section">
          {categories.map((category) => (
            <GiphyGrid key={category} category={category} />
          ))}
        </section>
      </main>
    </>
  );
};
