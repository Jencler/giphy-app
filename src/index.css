@import url("https://fonts.googleapis.com/css2?family=Geist:wght@100..900&display=swap");

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html {
  font-family: "Geist", sans-serif;
  font-optical-sizing: auto;
  font-style: normal;
  font-size: 62.5%;
}

body {
  width: 100%;
  height: 100%;
}

.header {
  width: 100%;
  height: 68px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1024px;
  margin-right: auto;
  margin-left: auto;
  padding: 0 16px;
}

.logo__container {
  width: 170px;
  height: 68px;
  display: flex;
  gap: 6px;
}

.main {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  max-width: 1024px;
  margin-right: auto;
  margin-left: auto;
  padding: 58px 16px;
  text-align: center;
}

.main__top__section {
  display: flex;
  flex-direction: column;
  gap: 16px;

  & h1 {
    font-size: 3.4rem;
    text-align: center;
  }
}

.main_middle-section {
  padding: 24px 0;

  & label {
    display: flex;
    flex-direction: column;
    gap: 8px;
    align-items: flex-start;
    font-size: 1.8rem;
  }

  & input {
    padding: 12px 16px;
    font-size: 1.8rem;
    border-radius: 6px;
    border: 2px solid #322a2a;
    outline: none;
    min-width: 260px;
  }
}

.main__giphy-section {
  width: 100%;
  height: auto;
  padding: 16px;
}

.cards-containers {
  width: 100%;
  padding: 20px 0;
  margin: 0 auto;
}

.grid-giphys {
  display: flex;
  flex-wrap: wrap;
  gap: 24px;
  justify-content: center;
  align-items: center;
  
}

.card-giphy {
  width: 100%;
  max-width: 200px;
  height: auto;
  display: flex;
  flex-direction: column;
  gap: 8px;
  align-items: flex-start;
  border-radius: 8px;
  padding: 12px;
  position: relative;
  border: 2px solid #444;
  box-shadow: 4px 4px 8px #44444432,-4px -4px 8px #4444442a ;

  & img {
    width: 100%;
    height: 100%;
    object-fit: contain;
  }
}

.subtitle {
  padding: 10px 0;
  font-size: 18px;
  text-transform: uppercase;
}

/* From Uiverse.io by mrhyddenn */
.spinner {
  width: 3em;
  height: 3em;
  cursor: not-allowed;
  border-radius: 50%;
  border: 2px solid #444;
  box-shadow: -10px -10px 10px #6359f8, 0px -10px 10px 0px #9c32e2,
    10px -10px 10px #f36896, 10px 0 10px #ff0b0b, 10px 10px 10px 0px#ff5500,
    0 10px 10px 0px #ff9500, -10px 10px 10px 0px #ffb700;
  animation: rot55 0.7s linear infinite;
  margin: 0 auto;
}

.spinnerin {
  border: 2px solid #444;
  width: 1.5em;
  height: 1.5em;
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

@keyframes rot55 {
  to {
    transform: rotate(360deg);
  }
}

