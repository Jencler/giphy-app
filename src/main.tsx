import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import { GiphytApp } from "./GiphytApp.tsx";
import "./index.css";

createRoot(document.getElementById("root")!).render(
  <StrictMode>
    <GiphytApp />
  </StrictMode>
);
