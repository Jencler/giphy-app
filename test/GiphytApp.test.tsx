import { fireEvent, render, screen } from "@testing-library/react";
import { GiphytApp } from "../src/GiphytApp";
describe("Probar el <GiphytApp/>", () => {
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GiphytApp />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Giphys Application ðŸŽ‰"));
  });
  test("Debe de hacer match con el snapshot", () => {
    render(<GiphytApp />);
    // Obtenemos los elementos
    const form = screen.getByRole("form");
    const input = screen.getByRole("textbox") as HTMLInputElement;

    // Usamos los eventos
    fireEvent.change(input, {
      target: {
        value: "Coco",
      },
    });

    expect(input.value).toBe("Coco");
    fireEvent.submit(form);
    expect(input.value).toBe("");

  });
});
