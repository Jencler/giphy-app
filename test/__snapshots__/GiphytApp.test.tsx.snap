// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el <GiphytApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="logo__container"
    >
      <svg
        class="sc-1f039a7e-2 bMXkCi"
        viewBox="0 0 28 35"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-rule="evenodd"
          fill-rule="evenodd"
        >
          <path
            d="M0 3h4v29H0z"
            fill="#00ff99"
          />
          <path
            d="M24 11h4v21h-4z"
            fill="#9933ff"
          />
          <path
            d="M0 31h28v4H0z"
            fill="#00ccff"
          />
          <path
            d="M0 0h16v4H0z"
            fill="#fff35c"
          />
          <path
            d="M24 8V4h-4V0h-4v12h12V8"
            fill="#ff6666"
          />
          <path
            class="shadow"
            d="M24 16v-4h4M16 0v4h-4"
          />
        </g>
      </svg>
      <svg
        class="sc-1f039a7e-0 cArwE"
        viewBox="0 0 128 36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <g>
            <path
              d="m22.6,12c-2,-1.9 -4.4,-2.4 -6.2,-2.4c-4.4,0 -7.3,2.6 -7.3,8c0,3.5 1.8,7.8 7.3,7.8c1.4,0 3.7,-0.3 5.2,-1.4l0,-3.5l-6.9,0l0,-6l13.3,0l0,12.1c-1.7,3.5 -6.4,5.3 -11.7,5.3c-10.7,0 -14.8,-7.2 -14.8,-14.3c0,-7.1 4.7,-14.4 14.9,-14.4c3.8,0 7.1,0.8 10.7,4.4l-4.5,4.4zm9.1,19.2l0,-27.2l7.6,0l0,27.2l-7.6,0zm20.1,-7.4l0,7.3l-7.7,0l0,-27.1l13.2,0c7.3,0 10.9,4.6 10.9,9.9c0,5.6 -3.6,9.9 -10.9,9.9l-5.5,0zm0,-6.5l5.5,0c2.1,0 3.2,-1.6 3.2,-3.3c0,-1.8 -1.1,-3.4 -3.2,-3.4l-5.5,0l0,6.7zm36.7,13.9l0,-10.3l-9.8,0l0,10.3l-7.7,0l0,-27.2l7.7,0l0,10.3l9.8,0l0,-10.3l7.6,0l0,27.2l-7.6,0zm24.2,-17.9l5.9,-9.3l8.7,0l0,0.3l-10.8,16l0,10.8l-7.7,0l0,-10.8l-10.3,-16l0,-0.3l8.7,0l5.5,9.3z"
              id="svg_2"
            />
          </g>
        </g>
      </svg>
    </div>
  </header>
  <main
    class="main"
  >
    <section
      class="main__top__section"
    >
      <h1>
        Giphys Application ðŸŽ‰
      </h1>
    </section>
    <section
      class="main_middle-section"
    >
      <form
        aria-label="form"
        role="form"
      >
        <label
          class="label"
        >
          <span>
            Agrega una categorÃ­a
          </span>
          <input
            autocomplete="off"
            placeholder="Happy, smile, love, ..."
            type="text"
            value=""
          />
        </label>
      </form>
    </section>
    <section
      class="main__giphy-section"
    />
  </main>
</div>
`;
